### PR TITLE
feat: migrate to webcomponents

### PR DESCRIPTION
**Pull request by BenjaminPabst**

### CODE REVIEW

File: nx-plugin/src/generators/angular/files/helm/values.yaml [hunk 1]
```diff
@@ -15,6 +15,10 @@ app:
         description: "OneCX <%= remoteModuleName %> Module"
         note: "OneCX <%= remoteModuleName %> UI module auto import via MF operator"
         type: MODULE
+        main:          
+          technology: WEBCOMPONENTMODULE
+          remoteName: <%= remoteModuleFileName %>
+          tagName: <%= remoteModuleFileName %>-webcomponent
```
Comment (markuczy) [inline nx-plugin/src/generators/angular/files/helm/values.yaml:21]: Looks different than for tenant-ui (which is basing on generator). Make sure its correct for whatever operator those should use (https://github.com/onecx/onecx-tenant-ui/commit/1586479ff7da5b391eb4f62e4d1680aab1f7b640#diff-81801117ec01136c8a49bfcd42af8e104f4efad1f2daccda9da9d960eaad5279R13)

File: nx-plugin/src/generators/angular/files/src/app/__remoteModuleFileName__-app.remote.module.ts.template [hunk 2]
```diff
@@ -23,6 +23,13 @@ import { Configuration } from './shared/generated';
 import { SharedModule } from './shared/shared.module';
 import { EffectSources, EffectsRunner, Actions  } from '@ngrx/effects';
 import { apiConfigProvider } from './shared/utils/apiConfigProvider.utils';
+import { DoBootstrap, Injector } from '@angular/core'
+import { createCustomElement } from '@angular/elements'
+import { Router } from '@angular/router'
+import { APP_INITIALIZER } from '@angular/core'
+import { initializeRouter } from '@onecx/angular-webcomponents'
+import { AppStateService } from '@onecx/angular-integration-interface'
+import { AppEntrypointComponent } from './app-entrypoint.component';
```
Comment (markuczy) [inline nx-plugin/src/generators/angular/files/src/app/__remoteModuleFileName__-app.remote.module.ts.template:32]: Did You add required packages somewhere? You should have:

- @angular/elements
- @onecx/angular-auth (most likely)
- @onecx/angular-webcomponents
- @webcomponents/webcomponentsjs
Maybe some more (not sure tho)

Comment (markuczy) [reply nx-plugin/src/generators/angular/files/src/app/__remoteModuleFileName__-app.remote.module.ts.template:32]: Also You need to add the `node_modules/@webcomponents/webcomponentsjs/webcomponents-bundle.js` to the build

File: nx-plugin/src/generators/angular/files/src/app/__remoteModuleFileName__-app.remote.module.ts.template [hunk 3]
```diff
@@ -66,6 +76,21 @@ effectProvidersForWorkaround.forEach(p => p.ɵprov.providedIn = null);
...
-export class <%= remoteModuleName %>Module {}
+export class <%= remoteModuleName %>Module implements DoBootstrap {
+  constructor(private injector: Injector) {}
+
+  ngDoBootstrap(): void {
+    const appEntrypoint = createCustomElement(AppEntrypointComponent, {
+      injector: this.injector
+    })
+    customElements.define('<%= remoteModuleFileName %>-webcomponent', appEntrypoint)
+  }
```
Comment (markuczy) [inline nx-plugin/src/generators/angular/files/src/app/__remoteModuleFileName__-app.remote.module.ts.template:95]: You should use new approach with createAppEntrypoint

File: nx-plugin/src/generators/angular/generator.ts [hunk 4]
```diff
@@ -87,7 +87,7 @@ export async function angularGenerator(
     tree.delete(`${directory}/scripts/load-permissions.sh`);
   }
 
-  const oneCXLibVersion = '^4.40.2';
+  const oneCXLibVersion = '^4.41.0';
```
Comment (markuczy) [inline nx-plugin/src/generators/angular/generator.ts:90]: Use 4.22.0

File: nx-plugin/src/generators/details/steps/feature-routes.step.ts [hunk 5]
```diff
@@ -2,23 +2,27 @@ import { Tree, names } from '@nx/devkit';
...
     let moduleContent = tree.read(routesFilePath, 'utf8');
-    moduleContent = moduleContent.replace(
-      'routes: Routes = [',
-      `routes: Routes = [ { path: 'details/:id', component: ${className}DetailsComponent, pathMatch: 'full' },`
-    );
+    moduleContent =
+      `import { startsWith } from '@onecx/angular-webcomponents'` +
+      moduleContent.replace(
+        'routes: Routes = [',
+        `routes: Routes = [ { matcher: startsWith('details/:id'), component: ${className}DetailsComponent, pathMatch: 'full' },`
```
Comment (markuczy) [inline nx-plugin/src/generators/details/steps/feature-routes.step.ts:17]: This most likely will not work :/

Comment (markuczy) [reply nx-plugin/src/generators/details/steps/feature-routes.step.ts:17]: You will need to create Your own matcher for this

**Review by markuczy — APPROVED** [review]

File: nx-plugin/src/generators/angular/files/webpack.config.js.template [hunk 6]
```diff
@@ -45,6 +46,10 @@ const config = withModuleFederationPlugin({
       requiredVersion: 'auto',
       includeSecondaries: true,
     },
+    '@onecx/angular-accelerator': {
```
Comment (SchettlerKoehler) [inline nx-plugin/src/generators/angular/files/webpack.config.js.template:49]: `@onecx/webcomponents` should also be here

File: nx-plugin/src/generators/delete/steps/search-tests.step.ts [hunk 7]
```diff
@@ -11,11 +11,13 @@ export class SearchTestsStep implements GeneratorStep<DeleteGeneratorSchema> {
 
     let htmlContent = tree.read(filePath, 'utf8');
 
-    htmlContent =
-      `import { PrimeIcons } from 'primeng/api';` +
-      htmlContent.replace(
-        "it('should dispatch export csv data on export action click'",
-        `
+    if (!htmlContent.includes(`import { PrimeIcons } from 'primeng/api';`)) {
```
Comment (SchettlerKoehler) [inline nx-plugin/src/generators/delete/steps/search-tests.step.ts:14]: We are always doing organize imports at the end, so duplicate imports will be removed

**Review by SchettlerKoehler — CHANGES_REQUESTED** [review]

File: nx-plugin/src/generators/delete/steps/search-tests.step.ts (hunk 7 above)
Comment (BenjaminPabst) [reply nx-plugin/src/generators/delete/steps/search-tests.step.ts]: had that weirdly crashing my tests once, but now it works again, so removed the check.

**Review by SchettlerKoehler — APPROVED** [review]

**Pull request merged**